tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/datatransfer/ImportExistingArchiveProjectFilterTest.java

```diff
@@ -155,6 +155,9 @@ private void processElementAndChildren(Object element, ITreeContentProvider cont
 			IFolder folder = (IFolder) element;
 			assertFalse(folder.getName().equalsIgnoreCase("res"));
 		} else {
+			if (element.getClass().getName().startsWith("org.eclipse.jdt")) {
```

Comment:
This is not optimal as e.g. org.eclipse.cdt can be installed on the machine and cause similar delay. Is it possible to "whitelist" rather than "blacklist" elements?

Reply:
i would prefer whitelist too. Just don't know what needs to be listed

Reply:
A sane way to figure would be to create launch configuration with the minimum plugins that o.e.ui.tests requires and print the elements to get such a list.

Reply:
I'm a tiny bit confused.  This is in a test in the Platform's I-build.  In what case would CDT being installed be a problem?

Reply:
I see.  You mean when we launch from the IDE and we might have CDT in the target platform.  So this is not concern for the I-Build itself (and in the IDE we can always assign more memory, though being slow is not nice either there).

Reply:
The change skips JDT contributions for the sake of speeding up tests, which is relevant for developers as the bundle itself doesn't require JDT thus tycho would not be sped up by this but only IDE runs of the test and developer can have any combination of plugins installed JDT, CDT, Mylyn, WildWeb,.... each of them slowing the test run. That's why I'm looking for whitelist (check only relevant aka Platform contributions or even smaller list) instead of blacklisting JDT.

Reply:
restricting IResource feels good and skips the same entries for me